# Project Description

## 1. Overview
Teaté is a digital platform with a B2B model that connects suppliers with small neighborhood grocery stores, which allows store owners the acquisition of products at convenient prices to supply their businesses. Store owners receive their supplies 48 hours after they order y  and own a digital wallet. Teaté believes in sustainable economic development, supporting small businesses and creating more collaborative and digital communities.

## 2. Problem Definition

Store owners have limited resources to supply their stores, and use their intuition to figure out which products h te q they should buy. This lack of information and  I  ppl small budget has an impact on their revenues and business, which affects the a al. Tl^’ possibility of offering a wider portfolio and satisfying demand.   

It is important to keep in mind that sales in the traditional channel represent 50% of total sales in mass consumer products, and lowering prices, will help them sell their products at competitive prices.

In order to maximize revenue and optimize resources for acquiring merchandise, the idea is to find optimal combinations between products regarding quantity and price to guarantee enough stock and avoid oversupply in the stores.

## 3. Datasets to use

The data sets for the project will be:

- Historical transactions of the app Teaté between stores and suppliers from January to July 2020. These transactions contain information about each product sold by the app.The main variables of the data set are related to the store (ID, location), the products (ID, quantity, cost) and also the dates of orders. 
- Public socio-economic data provided by DANE.



In [63]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

months = ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio', 'Julio']
list_of_df = []
for m in months:
    list_of_df.append(pd.read_csv('data/Teaté - Venta '+m+' 2020.csv', encoding='ISO-8859-1', sep=';', low_memory=False))

In [65]:
# Now lets concatenate our list of DFs
final_df = pd.concat(list_of_df)

final_df.head()

Pedido Fecha Pedido    Tienda   Nombre Tienda   Dirección Tienda  \
0  1000092736    2/01/2020  20009430  Tienda de memo  Cr 58 # 48 sur 15   
1  1000092736    2/01/2020  20009430  Tienda de memo  Cr 58 # 48 sur 15   
2  1000092736    2/01/2020  20009430  Tienda de memo  Cr 58 # 48 sur 15   
3  1000092736    2/01/2020  20009430  Tienda de memo  Cr 58 # 48 sur 15   
4  1000092736    2/01/2020  20009430  Tienda de memo  Cr 58 # 48 sur 15   

   Fabricante                  Nombre Fabricante  PosPed  Material  \
0     5000051                  DETERGENTES LTDA.     101       861   
1     5000039  JOHNSON & JOHNSON DE COLOMBIA S.A     602       382   
2     5000077               PRODUCTOS YUPI S.A.S     302       908   
3     5000079     PROCTER & GAMBLE COLOMBIA LTDA    1002       925   
4     5000039  JOHNSON & JOHNSON DE COLOMBIA S.A     601       382   

                        Nombre Material  ...  Unnamed: 46 Unnamed: 47  \
0     Detergente Dersa Liquido Doy Pack  ...          NaN         NaN   
1  JOHNSONS baby aceite  original 50 ML  ...          NaN         NaN   
2              Palomitas Caramelo 68 gr  ...          NaN         NaN   
3            Detergente Rindex 10 250gr  ...          NaN         NaN   
4  JOHNSONS baby aceite  original 50 ML  ...          NaN         NaN   

   Unnamed: 48 Zona Nombre ZonaComercial Ruta Nombre Ruta Comuna  Barrio  \
0          NaN  NaN                  NaN  NaN         NaN    NaN     NaN   
1          NaN  NaN                  NaN  NaN         NaN    NaN     NaN   
2          NaN  NaN                  NaN  NaN         NaN    NaN     NaN   
3          NaN  NaN                  NaN  NaN         NaN    NaN     NaN   
4          NaN  NaN                  NaN  NaN         NaN    NaN     NaN   

  Población  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN  

[5 rows x 56 columns]

In [66]:
final_df.to_csv('DF_completo.csv', index=False)

### Data Cleaning Process
Now lets see general information about our DataFrame, like its shape, number of non-null elements on each column and all unique values for some specific columns.

In [43]:
print('Our RAW DataFrame has',final_df.shape[0],'rows and',final_df.shape[1],'columns','\n')

Our RAW DataFrame has 325094 rows and 56 columns 



In [44]:
final_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 325094 entries, 0 to 55194
Data columns (total 56 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   Pedido                       325094 non-null  int64  
 1   Fecha Pedido                 325094 non-null  object 
 2   Tienda                       325094 non-null  int64  
 3   Nombre Tienda                325094 non-null  object 
 4   Dirección Tienda             325094 non-null  object 
 5   Fabricante                   325094 non-null  int64  
 6   Nombre Fabricante            325094 non-null  object 
 7   PosPed                       325094 non-null  int64  
 8   Material                     325094 non-null  int64  
 9   Nombre Material              325094 non-null  object 
 10  Cantidad de pedido           325094 non-null  int64  
 11  UM                           325094 non-null  object 
 12  Valor Unitario Pedido        325094 non-null  float64
 13  

In [45]:
len(final_df['Nombre Tienda'].apply(lambda x: x.lower()).unique())

6765

In [46]:
len(final_df['Nombre Tienda'].unique())

7299

In [47]:
len(final_df['Tienda'].unique())

9335

In [48]:
final_df['Nombre Tienda'] = final_df['Nombre Tienda'].apply(lambda x: x.lower())
final_df

Pedido Fecha Pedido    Tienda           Nombre Tienda  \
0      1000092736    2/01/2020  20009430          tienda de memo   
1      1000092736    2/01/2020  20009430          tienda de memo   
2      1000092736    2/01/2020  20009430          tienda de memo   
3      1000092736    2/01/2020  20009430          tienda de memo   
4      1000092736    2/01/2020  20009430          tienda de memo   
...           ...          ...       ...                     ...   
55190  1000189932   31/07/2020  20001962     tienda mixta yaneth   
55191  1000189992   31/07/2020  20001879            tienda mixta   
55192  1000190008   31/07/2020  20003029        variedades nicol   
55193  1000190024   31/07/2020  20011937  papeleria y variedades   
55194  1000190054   31/07/2020  20004287     tienda mixta martha   

                     Dirección Tienda  Fabricante  \
0                   Cr 58 # 48 sur 15     5000051   
1                   Cr 58 # 48 sur 15     5000039   
2                   Cr 58 # 48 sur 15     5000077   
3                   Cr 58 # 48 sur 15     5000079   
4                   Cr 58 # 48 sur 15     5000039   
...                               ...         ...   
55190            CL 101 E # 24 B - 04     5000057   
55191  AV 2 B2 # 73N-151 UN Residenci     5000057   
55192  CR 83 C # 30 - 08 CON B CAS 41     5000077   
55193                     CR 28#57-18     5000094   
55194                 CL 80 # 23 - 89     5000027   

                       Nombre Fabricante  PosPed  Material  \
0                      DETERGENTES LTDA.     101       861   
1      JOHNSON & JOHNSON DE COLOMBIA S.A     602       382   
2                   PRODUCTOS YUPI S.A.S     302       908   
3         PROCTER & GAMBLE COLOMBIA LTDA    1002       925   
4      JOHNSON & JOHNSON DE COLOMBIA S.A     601       382   
...                                  ...     ...       ...   
55190            MONDELEZ COLOMBIA S.A.S     400       772   
55191            MONDELEZ COLOMBIA S.A.S     100      1393   
55192               PRODUCTOS YUPI S.A.S     700       912   
55193         MASTERFOODS COLOMBIA LTDA.     201      1384   
55194                     CASA LUKER S.A     200       126   

                                Nombre Material  ...  Unnamed: 46 Unnamed: 47  \
0             Detergente Dersa Liquido Doy Pack  ...          NaN         NaN   
1          JOHNSONS baby aceite  original 50 ML  ...          NaN         NaN   
2                      Palomitas Caramelo 68 gr  ...          NaN         NaN   
3                    Detergente Rindex 10 250gr  ...          NaN         NaN   
4          JOHNSONS baby aceite  original 50 ML  ...          NaN         NaN   
...                                         ...  ...          ...         ...   
55190                         Chiclets  Surtido  ...          NaN         NaN   
55191       CLUB SOCIAL INTEGRAL(R)EXP24PK IPOD  ...          NaN         NaN   
55192             Copelia Panela+Areq+Coco 15gr  ...          NaN         NaN   
55193  Alimento para perro PEDIGREE Sobre Adult  ...          NaN         NaN   
55194                         Luker Amargo 125G  ...          NaN         NaN   

       Unnamed: 48   Zona  Nombre ZonaComercial Ruta Nombre Ruta     Comuna  \
0              NaN    NaN                   NaN  NaN         NaN        NaN   
1              NaN    NaN                   NaN  NaN         NaN        NaN   
2              NaN    NaN                   NaN  NaN         NaN        NaN   
3              NaN    NaN                   NaN  NaN         NaN        NaN   
4              NaN    NaN                   NaN  NaN         NaN        NaN   
...            ...    ...                   ...  ...         ...        ...   
55190          NaN  208.0  Zona Comerc Cali 208    V     Viernes  COMUNA 21   
55191          NaN  202.0  Zona Comerc Cali 202    M      Martes   COMUNA 2   
55192          NaN  212.0  Zona Comerc Cali 212    V     Viernes  COMUNA 17   
55193          NaN  307.0   Zona Comerc Med 307  

In [49]:
temp_df = final_df[['Tienda','Nombre Tienda']].copy()

In [50]:
grp_df = temp_df.groupby('Tienda')

In [51]:
shops_more_1 = []

for item in grp_df:
    if len(item[1]['Nombre Tienda'].unique()) > 1:
        shops_more_1.append([item[1]['Tienda'].unique()[0], 
                             len(item[1]['Nombre Tienda'].unique())])
        
shops_more_1

[[20001266, 2],
 [20002466, 2],
 [20002786, 2],
 [20003231, 2],
 [20003513, 2],
 [20004964, 2],
 [20005557, 2],
 [20005741, 2],
 [20005784, 2],
 [20005804, 2],
 [20006172, 2],
 [20006341, 2],
 [20006482, 2],
 [20006540, 2],
 [20006544, 2],
 [20006585, 2],
 [20006595, 2],
 [20006733, 2],
 [20006813, 2],
 [20006857, 2],
 [20006946, 2],
 [20006979, 2],
 [20007058, 2],
 [20007713, 2],
 [20007830, 2],
 [20007913, 2],
 [20008123, 2],
 [20008420, 2],
 [20008720, 2],
 [20008879, 2],
 [20008999, 2],
 [20009213, 2],
 [20009493, 2],
 [20009666, 2],
 [20009668, 2],
 [20009735, 2],
 [20009783, 2],
 [20009789, 2],
 [20009944, 2],
 [20010164, 2],
 [20010347, 2],
 [20010370, 2],
 [20010562, 2],
 [20010854, 2],
 [20011047, 2],
 [20011201, 2],
 [20011234, 2],
 [20011329, 2],
 [20011333, 2],
 [20011545, 2],
 [20011558, 2],
 [20011629, 3],
 [20011654, 2],
 [20011723, 2],
 [20011767, 2],
 [20011777, 2],
 [20011994, 2],
 [20012318, 2],
 [20012328, 2],
 [20012470, 2],
 [20012491, 2],
 [20012588, 2],
 [200127

In [52]:
len(shops_more_1)

67

In [53]:
list(temp_df[temp_df['Tienda'] == shops_more_1[3][0]]['Nombre Tienda'].unique())

['panaderia sabrospan', 'tienda mixta tito']

In [54]:
pairs_of_shops = []
for i in range(0,len(shops_more_1)):
    pairs_of_shops.append(list(temp_df[temp_df['Tienda'] == shops_more_1[i][0]]['Nombre Tienda'].unique()))
    
pairs_of_shops

[['tienda a&j', 'tienda jj'],
 ['rapitienda lemar', 'rapitienda janeth'],
 ['tienda la esquina', 'tienda mixta'],
 ['panaderia sabrospan', 'tienda mixta tito'],
 ['tienda el viejo ping', 'fruver'],
 ['el plan de las 3 niñas', 'el plan de las 3 ninas'],
 ['tienda maria', 'tienda ahorra max'],
 ['viveres doña ana', 'viveres dona ana'],
 ['carpa azul número 2', 'carpa azul numero 2'],
 ['wreja café', 'wreja cafe'],
 ['la viña', 'la vina'],
 ['granero el caleño', 'granero el caleno'],
 ['tenda divino niño', 'tenda divino nino'],
 ['variedades y papelería encanto', 'variedades y papeleria encanto'],
 ['tienda el plan número 1', 'tienda el plan numero 1'],
 ['legumbrería la mazorca', 'legumbreria la mazorca'],
 ['charcutería y licores alba y jaime', 'charcuteria y licores alba y jaime'],
 ['tienda panadería la mona', 'tienda panaderia la mona'],
 ['granero mixto el pedrero', 'granero mixto el predrero'],
 ['él social', 'el social'],
 ['la economía', 'la economia'],
 ['charcutería galeano', '

In [55]:
[2,3]+[5]

[2, 3, 5]

In [56]:
pairs_of_shops_code = []
for i in range(0,len(shops_more_1)):
    pairs_of_shops_code.append(list(temp_df[temp_df['Tienda'] == shops_more_1[i][0]]['Nombre Tienda'].\
                                    unique())+shops_more_1[i])
    
pairs_of_shops_code

[['tienda a&j', 'tienda jj', 20001266, 2],
 ['rapitienda lemar', 'rapitienda janeth', 20002466, 2],
 ['tienda la esquina', 'tienda mixta', 20002786, 2],
 ['panaderia sabrospan', 'tienda mixta tito', 20003231, 2],
 ['tienda el viejo ping', 'fruver', 20003513, 2],
 ['el plan de las 3 niñas', 'el plan de las 3 ninas', 20004964, 2],
 ['tienda maria', 'tienda ahorra max', 20005557, 2],
 ['viveres doña ana', 'viveres dona ana', 20005741, 2],
 ['carpa azul número 2', 'carpa azul numero 2', 20005784, 2],
 ['wreja café', 'wreja cafe', 20005804, 2],
 ['la viña', 'la vina', 20006172, 2],
 ['granero el caleño', 'granero el caleno', 20006341, 2],
 ['tenda divino niño', 'tenda divino nino', 20006482, 2],
 ['variedades y papelería encanto',
  'variedades y papeleria encanto',
  20006540,
  2],
 ['tienda el plan número 1', 'tienda el plan numero 1', 20006544, 2],
 ['legumbrería la mazorca', 'legumbreria la mazorca', 20006585, 2],
 ['charcutería y licores alba y jaime',
  'charcuteria y licores alba y 

##### Now lets solve some problems with names.
The ones that are easier to solve are those that have some little mistakes like ['wreja café', 'wreja cafe'], this is because in this cases we could be sure that we are talking about the same store, so lets identify them individualy.

- ['el plan de las 3 niñas', 'el plan de las 3 ninas']
- ['viveres doña ana', 'viveres dona ana']
- ['carpa azul número 2', 'carpa azul numero 2']
- ['wreja café', 'wreja cafe']
- ['la viña', 'la vina']
- ['granero el caleño', 'granero el caleno'],
- ['tenda divino niño', 'tenda divino nino'],
- ['variedades y papelería encanto', 'variedades y papeleria encanto'],
- ['tienda el plan número 1', 'tienda el plan numero 1'],
- ['legumbrería la mazorca', 'legumbreria la mazorca'],
- ['charcutería y licores alba y jaime', 'charcuteria y licores alba y jaime'],
- ['tienda panadería la mona', 'tienda panaderia la mona'],
- ['granero mixto el pedrero', 'granero mixto el predrero'],
- ['él social', 'el social'],
- ['la economía', 'la economia'],
- ['charcutería galeano', 'charcuteria galeano'],
- ['don tomás', 'don tomas'],
- ['granero y legumbres guarniciones f', 'granero y legumbres guarniciones fy'],
- ['3052387218', 'legumbreria melani'],
- ['tienda ancaba', 'tienda ancava'],
- ['el buen  precio', 'el buen precio'],
- ['reha cafe', 'reja cafe'],
- ['mercados la 80', 'mercados la 83'],
- ['punto soi', 'punto sol'],
- ['granero la bendición', 'granero la bendicion'],
- ['droguería tejarez', 'droguerã\xada tejarez'],
- ['panadería del parque', 'panaderã\xada del parque'],
- ['miscelánea y tienda adriana', 'miscelã¡nea y tienda adriana'],
- ['tienda maría leonor', 'tienda marã\xada leonor'],
- ['panadería y restaurante deli pollo', 'panaderã\xada y restaurante deli pollo'],
- ['súper tienda la mejor', 'sãºper tienda la mejor'],
- ['miscelánea el vecino', 'miscelã¡nea el vecino'],
- ['eco farma droguerías', 'eco farma droguerã\xadas'],
- ['miscelánea yami', 'miscelã¡nea yami'],
- ['tienda mixta bolaños', 'tienda mixta bolaã±os'],
- ['tienda la economía', 'tienda la economã\xada'],
- ['tienda el trébol', 'tienda el trã©bol'],
- ['miscelánea y papelería campo alegre','miscelã¡nea y papelerã\xada campo aleg'],
- ['perfumería acqua', 'perfumerã\xada acqua'],
- ['miscelánea beltran', 'miscelã¡nea beltran'],
- ['rapitienda móvil la 55', 'rapitienda mã³vil la 55'],
- ['dónde vivi', 'dã³nde vivi'],
- ['variedadesskfc', 'variedades skfc'],
- ['la economía', 'la economã\xada']]

In [58]:
# Now lets re-run the previous code to verify that all names have been corrected and to identify those
# that still having the problem.
temp_df = final_df[['Tienda','Nombre Tienda']].copy()

grp_df = temp_df.groupby('Tienda')
shops_more_1 = []

for item in grp_df:
    if len(item[1]['Nombre Tienda'].unique()) > 1:
        shops_more_1.append([item[1]['Tienda'].unique()[0], 
                             len(item[1]['Nombre Tienda'].unique())])
        
shops_more_1

[[20001266, 2],
 [20002466, 2],
 [20002786, 2],
 [20003231, 2],
 [20003513, 2],
 [20004964, 2],
 [20005557, 2],
 [20005741, 2],
 [20005784, 2],
 [20005804, 2],
 [20006172, 2],
 [20006341, 2],
 [20006482, 2],
 [20006540, 2],
 [20006544, 2],
 [20006585, 2],
 [20006595, 2],
 [20006733, 2],
 [20006813, 2],
 [20006857, 2],
 [20006946, 2],
 [20006979, 2],
 [20007058, 2],
 [20007713, 2],
 [20007830, 2],
 [20007913, 2],
 [20008123, 2],
 [20008420, 2],
 [20008720, 2],
 [20008879, 2],
 [20008999, 2],
 [20009213, 2],
 [20009493, 2],
 [20009666, 2],
 [20009668, 2],
 [20009735, 2],
 [20009783, 2],
 [20009789, 2],
 [20009944, 2],
 [20010164, 2],
 [20010347, 2],
 [20010370, 2],
 [20010562, 2],
 [20010854, 2],
 [20011047, 2],
 [20011201, 2],
 [20011234, 2],
 [20011329, 2],
 [20011333, 2],
 [20011545, 2],
 [20011558, 2],
 [20011629, 3],
 [20011654, 2],
 [20011723, 2],
 [20011767, 2],
 [20011777, 2],
 [20011994, 2],
 [20012318, 2],
 [20012328, 2],
 [20012470, 2],
 [20012491, 2],
 [20012588, 2],
 [200127

In [59]:
len(shops_more_1)

67

So now we only have 23 stores that still having a problem with the name

In [60]:
pairs_of_shops = []
for i in range(0,len(shops_more_1)):
    pairs_of_shops.append(list(temp_df[temp_df['Tienda'] == shops_more_1[i][0]]['Nombre Tienda'].unique()))
    
pairs_of_shops

[['tienda a&j', 'tienda jj'],
 ['rapitienda lemar', 'rapitienda janeth'],
 ['tienda la esquina', 'tienda mixta'],
 ['panaderia sabrospan', 'tienda mixta tito'],
 ['tienda el viejo ping', 'fruver'],
 ['el plan de las 3 niñas', 'el plan de las 3 ninas'],
 ['tienda maria', 'tienda ahorra max'],
 ['viveres doña ana', 'viveres dona ana'],
 ['carpa azul número 2', 'carpa azul numero 2'],
 ['wreja café', 'wreja cafe'],
 ['la viña', 'la vina'],
 ['granero el caleño', 'granero el caleno'],
 ['tenda divino niño', 'tenda divino nino'],
 ['variedades y papelería encanto', 'variedades y papeleria encanto'],
 ['tienda el plan número 1', 'tienda el plan numero 1'],
 ['legumbrería la mazorca', 'legumbreria la mazorca'],
 ['charcutería y licores alba y jaime', 'charcuteria y licores alba y jaime'],
 ['tienda panadería la mona', 'tienda panaderia la mona'],
 ['granero mixto el pedrero', 'granero mixto el predrero'],
 ['él social', 'el social'],
 ['la economía', 'la economia'],
 ['charcutería galeano', '

In [61]:
# In the case of ['tienda a&j', 'tienda jj'], if we filter our DF by "Nombre Tienda" with each of them, individually
# 'Tienda a&j' only has a 3 registers in the same day and 'tienda jj' has more than 200 registers so lets replace tienda a&j
# with 'tienda jj'.

final_df['Nombre Tienda'].replace('tienda a&j', 'tienda jj', inplace=True)

In [62]:
final_df[final_df['Tienda'] == 20001266]

Pedido Fecha Pedido    Tienda Nombre Tienda     Dirección Tienda  \
38241  1000102688   28/01/2020  20001266     tienda jj  CR 24C   # 42A - 66   
38242  1000102688   28/01/2020  20001266     tienda jj  CR 24C   # 42A - 66   
38243  1000102688   28/01/2020  20001266     tienda jj  CR 24C   # 42A - 66   
27084  1000110580   18/02/2020  20001266     tienda jj  CR 24C   # 42A - 66   
27085  1000110580   18/02/2020  20001266     tienda jj  CR 24C   # 42A - 66   
27086  1000110580   18/02/2020  20001266     tienda jj  CR 24C   # 42A - 66   
27975  1000110879   18/02/2020  20001266     tienda jj  CR 24C   # 42A - 66   
27976  1000110879   18/02/2020  20001266     tienda jj  CR 24C   # 42A - 66   
27977  1000110879   18/02/2020  20001266     tienda jj  CR 24C   # 42A - 66   
27978  1000110879   18/02/2020  20001266     tienda jj  CR 24C   # 42A - 66   
34588  1000114080   24/02/2020  20001266     tienda jj  CR 24C   # 42A - 66   
1344   1000116973    2/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
1345   1000116973    2/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
1346   1000116973    2/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
1347   1000116973    2/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
3424   1000117881    4/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
3425   1000117881    4/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
26953  1000128384   24/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
26954  1000128384   24/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
26955  1000128384   24/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
26956  1000128384   24/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
26957  1000128384   24/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
26958  1000128384   24/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
26959  1000128384   24/03/2020  20001266     tienda jj  CR 24C   # 42A - 66   
8009   1000134952    6/04/2020  20001266     tienda jj  CR 24C   # 42A - 66   
8010   1000134952    6/04/2020  20001266     tienda jj  CR 24C   # 42A - 66   
8011   1000134952    6/04/2020  20001266     tienda jj  CR 24C   # 42A - 66   
10122  1000148652   11/05/2020  20001266     tienda jj  CR 24C   # 42A - 66   
10123  1000148652   11/05/2020  20001266     tienda jj  CR 24C   # 42A - 66   
10124  1000148652   11/05/2020  20001266     tienda jj  CR 24C   # 42A - 66   
10125  1000148652   11/05/2020  20001266     tienda jj  CR 24C   # 42A - 66   
10126  1000148652   11/05/2020  20001266     tienda jj  CR 24C   # 42A - 66   
10127  1000148652   11/05/2020  20001266     tienda jj  CR 24C   # 42A - 66   
10128  1000148652   11/05/2020  20001266     tienda jj  CR 24C   # 42A - 66   
16220  1000150954   18/05/2020  20001266     tienda jj  CR 24C   # 42A - 66   
1774   1000157525    2/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
1775   1000157525    2/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
1776   1000157525    2/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
9989   1000160914    8/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
9990   1000160914    8/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
20041  1000165192   16/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
20042  1000165192   16/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
20043  1000165192   16/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
20044  1000165192   16/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
20045  1000165192   16/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
25848  1000167768   23/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
25849  1000167768   23/06/2020  20001266     tienda jj  CR 24C   # 42A - 66   
52832  1000188614   30/07/2020  20001266     tienda jj  CR 24C   # 42A - 66   
52833  1000188614   30/07/2020  20001266     tienda jj  CR 24C   # 42A - 66   
52834  1000188614   30/07/2020  20001266     tienda jj  CR 24

Rapitienda Janeth: CL 9 # 53 A SUR - 04 - ID: 20002466

Rapitienda Lemar: CR 27 # 83 - 26 - ID: 20002466

final_df[final_df['Tienda'] == 20009953]

final_df[final_df['Tienda'] == 20008273]